### PR TITLE
tests for parent, name and mixins

### PR DESCRIPTION
**Pull request by bartlomieju**
ref #2872

### CODE REVIEW

File: test/unit/features/options/mixins.spec.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        b: function () {}
+      }
+    }).$mount()
+
+    expect(vm.a).toBeDefined()
+    expect(vm.b).toBeDefined()
+    expect(vm.$options.directives.c).toBeDefined()
+  })
+
+  it('should call hooks from mixins first, mixin methods should not override defined method', () => {
```
Comment (yyx990803) [inline test/unit/features/options/mixins.spec.js]: Can we split this into two separate cases? One for hook order and the other for method overriding.

File: test/unit/features/options/mixins.spec.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import Vue from 'vue'
+const mergeOptions = Vue.util.mergeOptions
+
+describe('Mixins', () => {
```
Comment (yyx990803) [inline test/unit/features/options/mixins.spec.js]: Let's prefix option test suites with Options. So it would be `describe('Options mixin', ...)`

**Comment by yyx990803** [comment]
Thanks for the PR! Just a few small things to be adjusted and LGTM after that.

**Comment by bartlomieju** [comment]
Adjusted to your guidelines 👍

**Pull request merged**